### ISSUE TITLE
[Enhancement]: Manage configuration for AWS-created log groups

### ISSUE THREAD

**Issue by patstrom** [hashicorp/terraform-provider-aws#27124]
### Description

Certain AWS resources create log groups with names that are known only after the apply has happened. This causes problems if you want to manage the configuration of these log groups using Terraform.

For example when creating a RabbitMQ cluster using `aws_mq_broker` one can enable logging by using the `logs` parameter. This will create some Cloudwatch log groups that are named after the broker id. I want to apply retention periods (and other configuration) to these log groups but since they are not created via Terraform, and I cannot create them ahead of time (as you can for this situation with a Lambda). I need to let the `apply` command return an error, import the resources and then continue.

Perhaps it would be possible to add a `cloudwatch_log_group_configuration`  resource that only manages configuration for an already existing log group, similar to the division on the s3 resource? Or does a solution to get around this problem exist already?

### Affected Resource(s) and/or Data Source(s)

* aws_cloudwatch_log_group`

### Potential Terraform Configuration

```terraform
resource "aws_mq_broker" "rabbitmq" {
    broker_name = "example"
    engine_type = "RabbitMQ"
    engine_version = "3.8.23"
    host_instance_type = "mq.t3.micro"
    user {
        username = "admin"
        password = "admin"
    }
    logs {
        general = true
    }
}

# This will be an error since the log groups are already created.
# They need to be imported before "apply" can continue
resource "aws_cloudwatch_log_group" "rabbitmq_loggroup" {
    for_each = toset(["channel", "connection", "federation", "general", "mirroring", "queue", "upgrade"])

    name = "/aws/amazonmq/broker/${aws_mq_broker.rabbitmq.id}/${each.key}"
    retention_in_days = 14
}

# This would manage configuration for log groups created outside of Terraform, which would mean one "apply" is enough.
resource "aws_cloudwatch_log_group_configuration" "rabbitmq_loggroup" {
    for_each = toset(["channel", "connection", "federation", "general", "mirroring", "queue", "upgrade"])

    name = "/aws/amazonmq/broker/${aws_mq_broker.rabbitmq.id}/${each.key}"
    retention_in_days = 14

    # would also support stuff like kms and whatnot
}
```

**Comment by jaceq** [comment]
Hi @patstrom so in first issue of yours, you can add 'depends_on = [aws_mq_broker.rabbitmq]' to resource "aws_cloudwatch_log_group" "rabbitmq_loggroup"
However I fully support request for adding: aws_cloudwatch_log_group_configuration that will allow configuration of pre-existing log groups

**Comment by patstrom** [comment]
That doesn't resolve my issue. The problem here is that when creating an aws_mq_broker it'll automatically create log groups that are named after the broker id. I have no option to force the aws_mq_broker to use any particular log group. Thus, I cannot create the log group ahead of time.

**Comment by chrisghill** [comment]
Yes, the issue seems to be that the cloudwatch log group name is based on the ID of the MQ broker. If the cloudwatch log group name was configurable, or if it was at least deterministic based on something other than a computed value (like the broker name instead of the ID) then the cloudwatch log group could be created first.

**Comment by marceloboeira** [comment]
I got the same issue, in my case it's mostly that the default logs group doesn't include the tags from the main resource, thus making it almost impossible to find the logs downstream only with the broker.id ...
